src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -0,0 +1,179 @@
+		final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+		ValidationResult  result = config.validate();
+
+		if(result.getErrors().size() > 0){
+			throw new PermanentEventNotificationException("Please verify your webhook URL");
+		}
+
+		try {
+			SlackMessage slackMessage = createSlackMessage(ctx, config);
+			slackClient.send(slackMessage,config.webhookUrl());
+		} catch (Exception e) {
+			final Notification systemNotification = notificationService.buildNow()
+					.addNode(nodeId.toString())
+					.addType(Notification.Type.GENERIC)
+					.addSeverity(Notification.Severity.NORMAL)
+					.addDetail("exception", e.toString());
+
+			notificationService.publishIfFirst(systemNotification);
+			throw new TemporaryEventNotificationException("Slack notification is triggered, but sending failed. " + e.getMessage(), e);
+
+		}
+
+	}
+
+	/**
+	 *
+	 * @param ctx
+	 * @param config
+	 * @return
+	 * @throws PermanentEventNotificationException - throws this exception when the custom message template is invalid
+	 */
+	SlackMessage createSlackMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) throws PermanentEventNotificationException {
+		//Note: Link names if notify channel or else the channel tag will be plain text.
+		boolean linkNames = config.linkNames() || config.notifyChannel();
+		String message = buildDefaultMessage(ctx, config);
+
+		String customMessage = null;
+		String template = config.customMessage();
+		if (!isNullOrEmpty(template)) {
+			customMessage = buildCustomMessage(ctx, config, template);
+		}
+
+		return new SlackMessage(
+				config.color(),
+				config.iconEmoji(),
+				config.iconUrl(),
+				config.userName(),
+				config.channel(),
+				linkNames,
+				message,
+				customMessage
+				);
+	}
+
+	String buildDefaultMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) {
+		String title = buildMessageTitle(ctx);
+
+		// Build custom message
+		String audience = config.notifyChannel() ? "@channel " : "";
+		String description = ctx.eventDefinition().map(EventDefinitionDto::description).orElse("");
+		return String.format(Locale.ROOT,"%s*Alert %s* triggered:\n> %s \n", audience, title, description);
+	}
+
+	private String buildMessageTitle(EventNotificationContext ctx) {
+		String eventDefinitionName = ctx.eventDefinition().map(EventDefinitionDto::title).orElse("Unnamed");
+		return "_" + eventDefinitionName + "_";
+	}
+
+	String buildCustomMessage(EventNotificationContext ctx, SlackEventNotificationConfig config, String template) throws PermanentEventNotificationException {
+		List<MessageSummary> backlog = getAlarmBacklog(ctx);
+		Map<String, Object> model = getCustomMessageModel(ctx, config, backlog);
+		try {
+			LOG.debug("template = {} model = {}" ,template, model);
+			return templateEngine.transform(template, model);
+		} catch (Exception e) {
+			LOG.error("Exception during templating", e);
```

Comment:
When you pass the Exception as an argument to `LOG.error()`, it will print the stack trace in the log.  As a general rule, we should only print the stack trace if we're not passing the exception back up the call chain because some method further up the call chain is also likely to print the stack trace and having multiple copies of the stack trace in the log is both inefficient and confusing for users.  Since this exception is going to be re-thrown, we should not print the stack trace.

```suggestion
			LOG.error("Exception during templating [{}]", e.toString());
```